### PR TITLE
Add 'Topic.test_iam_permissions' API wrapper.

### PR DESCRIPTION
**Pull request by tseaver**
~~Uses #1641 as a base.~~

Toward #1073.

### CODE REVIEW

**Comment by tseaver** [comment]
@dhermes Rebased after merging #1641 to `master`.  PTAL

File: gcloud/pubsub/topic.py [hunk 1]
```diff
@@ -303,6 +303,31 @@ def set_iam_policy(self, policy, client=None):
             method='POST', path=path, data=resource)
         return Policy.from_api_repr(resp)
 
+    def test_iam_permissions(self, permissions, client=None):
```
Comment (dhermes) [inline gcloud/pubsub/topic.py]: I realize the method is called `testIamPermissions` but wouldn't `check_iam_permissions` be a better name?

**Comment by dhermes** [comment]
LGTM other than name issue

File: gcloud/pubsub/topic.py (hunk 1 above)
Comment (tseaver) [inline gcloud/pubsub/topic.py]: That would be a reasonable choice:  I'm on the fence about whether the improvement would be worth drifting from the method name used in the API.

Comment (tseaver) [inline gcloud/pubsub/topic.py]: Opened #1648 and assigned to @jgeewax for the tie-breaking vote.

Comment (jgeewax) [inline gcloud/pubsub/topic.py]: I'm a fan of `check_*` as it's more Pythonic.... We can always alias it if people become frustrated... I think the API method is the one that should change to `checkIamPermissions`

**Pull request merged**